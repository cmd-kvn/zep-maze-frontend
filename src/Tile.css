:local(.wall) {
  background-color: #696969;
  border: 4px outset #a9a9a9; }

:local(.floor) {
  background-color: #C0C0C0;
  border: 4px outset #a9a9a9; }

:local(.start) {
  background-color: #FFFF00;
  border: 4px outset #a9a9a9; }

:local(.end) {
  background-color: #0000FF;
  border: 4px outset #a9a9a9; }
