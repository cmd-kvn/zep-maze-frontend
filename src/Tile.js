import React from 'react';

export function Tile({ value }) {


    return (
        <td></td>
    )
}