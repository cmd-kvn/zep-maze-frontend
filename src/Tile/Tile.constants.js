export const FLOOR = 'FLOOR';
export const WALL = 'WALL';
export const START = 'START';
export const END = 'END';