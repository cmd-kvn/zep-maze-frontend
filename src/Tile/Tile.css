:local(.wall) {
  height: 32;
  width: 32;
  background-color: #696969;
  border: 4px outset #a9a9a9; }

:local(.floor) {
  height: 32;
  width: 32;
  background-color: #C0C0C0;
  border: 4px outset #a9a9a9; }

:local(.start) {
  height: 32;
  width: 32;
  background-color: #FFFF00;
  border: 4px outset #a9a9a9; }

:local(.end) {
  height: 32;
  width: 32;
  background-color: #0000FF;
  border: 4px outset #a9a9a9; }
