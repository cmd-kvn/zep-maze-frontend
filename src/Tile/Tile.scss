$size: 48px;

:local(.tile){
    height: $size;
    width: $size;
    box-sizing: border-box;
    border: 4px outset #a9a9a9;
    text-align: center;
    vertical-align: middle;
    &.wall{
        background-color: #696969;
        &:before{
            background-image: "ðŸœ˜";
        }
    }
    &.floor{
        background-color: #C0C0C0;
    }
    &.start{
        background-color: #FFFF00;
        &:before{
            content: "ðŸœ˜";
        }
    }
    &.end{
        background-color: #0000FF;
    }
}