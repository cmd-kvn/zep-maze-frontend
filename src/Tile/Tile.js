import React from 'react';
import { FLOOR, WALL, START, END } from './Tile.constants';
import styles from './Tile.css';

export function Tile({ value = '' }) {
    return (
        <td className={`${styles.tile} ${value.toLowerCase()}`}>
        </td>
    );
}