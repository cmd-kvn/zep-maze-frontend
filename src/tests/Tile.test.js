import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Tile } from '../Tile/Tile';
import { WALL, FLOOR, START, END } from '../Tile/Tile.constants';

describe('Tile component', () => {

    it('renders an empty td element by default', () => {
        const wrapper = shallow(<Tile/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    })
    
    describe('provides appropriate styling when passed a value', () => {
        it('wall styling', () => {
            const wrapper = shallow(<Tile value={WALL}/>);
            expect(toJSON(wrapper)).toMatchSnapshot();
        });

        it('floor styling', () => {
            const wrapper = shallow(<Tile value={FLOOR}/>);
            expect(toJSON(wrapper)).toMatchSnapshot();
        });

        it('Start styling', () => {
            const wrapper = shallow(<Tile value={START}/>);
            expect(toJSON(wrapper)).toMatchSnapshot();
        });

        it('end styling', () => {
            const wrapper = shallow(<Tile value={END}/>);
            expect(toJSON(wrapper)).toMatchSnapshot();
        });

    })
})