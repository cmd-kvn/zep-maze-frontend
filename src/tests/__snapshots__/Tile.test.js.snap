// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile component provides appropriate styling when passed a value Start styling 1`] = `
<td>
  START
</td>
`;

exports[`Tile component provides appropriate styling when passed a value end styling 1`] = `
<td>
  END
</td>
`;

exports[`Tile component provides appropriate styling when passed a value floor styling 1`] = `<td />`;

exports[`Tile component provides appropriate styling when passed a value wall styling 1`] = `<td />`;

exports[`Tile component renders an empty td element by default 1`] = `<td />`;
